### DISCUSSION TITLE
HA Companion e Fritzbox

### DISCUSSION THREAD

**Discussion by popeye70** [indomus/forum#2430] [HUB: Home Assistant]
Ciao a tutti,
ho una connessione internet di Fastweb con IP fisso, e un modem/router Fritzbox 7590 a cui dall'esterno accedo con l'indirizzo https://miorouter.myfritz.net:45618, utilizza un certificato gratuito di letsencrypt.org.
Non avendo bisogno di un DDNS esterno non ho quindi adottato DuckDNS, e al momento la mia app di HA sta accedendo dall'url http://miorouter.myfritz.net:8123 (ho abilitato quella porta).
Vorrei però accedere col protocollo https, è fattibile/consigliabile farlo utilizzando il certificato generato dal Fritz (si tratta di un .cer senza password)? Questo certificato ha una scadenza e andrà quindi periodicamente aggiornato?
Nel caso, mi confermate che la procedura è:

1) salvare il file boxcert.cer nel subfolder "ssl" della cartella homeassistant;

2) aggiungere al configuration.yaml la riga
http:
  ssl_certificate: /ssl/boxcert.cer

3) andare ad inserire nel campo “Configurazione” > “Sistema” > “Rete” e nel box “URL di Home Assistant” > “Internet” l'indirizzo https://miorouter.myfritz.net:8123

E' tutto qui, oppure mi sono perso qualche passaggio?
Grazie e saluti.

**Comment by Palmus84** [comment]
Uso anche io HA e FritzBox (7590ax), xò con Caddy: penso che i certificati li gestisca lui..

**Comment by lux90ita** [comment]
Hai più strade da percorrere
1 - usi la VPN del Fritz con i cellulari metodo più sicuro e ti colleghi (lasciando tutte le porte chiuse)
2 - lasci la porta 8123 e ti colleghi in http
3 - usi Caddy con la porta che usi https aperta esempio delle configurazioni se non sbaglio (9000) e ogni 3 mesi devi aprire le porte 443 e 80 per poi richiuderle (crei un automatismo che le apre e chiude in automatico allo scadere del certificato)

se conoscete altri metodi scrivete pure ;D
